src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -0,0 +1,103 @@
+package frc.robot.commands;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.subsystems.gpm_subsystem.Outtake;
+import frc.robot.subsystems.gpm_subsystem.Wrist;
+
+/**
+ * Shoots on the move.
+ */
+public class Shoot extends Command {
+
+    private static final Pose3d SPEAKER_POSE = new Pose3d(0, 0, 2.055, new Rotation3d());
```

Comment:
Use the poses in VisionConstants.

Reply:
does the z value on each speaker pose equal to 2.055? ben made the physics calculations and all the math stuff with a height of 2.055 in mind.

Reply:
If it does, it probably shouldn't since the shooter height is no longer always at 2.055.

Reply:
The speaker is at a constant height, but it is not at (0, 0). VisionConstants has the x, y, and z coordinates, and you usually shouldn't define constants directly in commands. In initialize or execute, you can define a variable similar to this:
Pose3d speakerPose = DriverStation.getAlliance().get()==Alliance.Blue?VisionConstants.BLUE_SPEAKER_POSE:VisionConstants.RED_SPEAKER_POSE;
For the calculations, you can get the shooter's pose relative to the speaker, including the relative height, and the calculations should still work.

Reply:
alright